spec/features/guests/show_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+
+RSpec.describe 'show view' do
+  let!(:holliday) { Hotel.create!(name: 'Holliday Inn', location: 'Buxton') }
+  let!(:harbor) { Hotel.create!(name: 'Portland Harbor', location: 'Portland') }
+  let!(:room_1) { harbor.rooms.create!(rate: 180, suite: "Presidential")}
+  let!(:room_2) { harbor.rooms.create!(rate: 185, suite: "Vice-Presidential")}
+  let!(:room_3) { holliday.rooms.create!(rate: 160, suite: "Honeymoon")}
+  let!(:room_4) { holliday.rooms.create!(rate: 10, suite: "Broom Closet")}
+  let!(:steve) { Guest.create!(name: 'Steve', nights: 3) }
+  let!(:dave) { Guest.create!(name: 'Dave', nights: 5) }
+  let!(:jim) { Guest.create!(name: 'Jim', nights: 1)}
+
+  before do
+    GuestRoom.create!(guest: steve, room: room_1)
+    GuestRoom.create!(guest: steve, room: room_2)
+    GuestRoom.create!(guest: dave, room: room_3)
+    GuestRoom.create!(guest: dave, room: room_4)
+  end
+
+  it 'displays list of all the rooms they have stayed in with with all attributes' do
+
+    visit "/guests/#{steve.id}"
+
+    expect(page).to have_content(steve.name)
+    expect(page).to have_content(room_1.suite)
+    expect(page).to have_content(room_1.rate)
+    expect(page).to have_content(room_2.suite)
+    expect(page).to have_content(room_2.rate)
+    expect(page).to have_content(harbor.name)
+
+    visit "/guests/#{dave.id}"
+
+    expect(page).to have_content(dave.name)
+    expect(page).to have_content(room_3.suite)
+    expect(page).to have_content(room_3.rate)
+    expect(page).to have_content(holliday.name)
+  end
```

Comment:
I believe, in order to make your rests a little more robust, to do some interpolation on what you are expecting on the page. Example:

```ruby
expect(page).to have_content("#{room_1.suite} #{room_1.id} ($#{ room.rate} at the #{room.hotel.name})")
```

This way we are ensured that we are looking at the right content and that `room_1.rate` that we are expecting to be on the page isn't just some random number that RSpec so on the page as maybe part of a date or something like that and marked as being found.